elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -46,50 +53,66 @@
     private final QueryRunner runner;
     private AsyncQueryDAO asyncQueryDao;
     private String apiVersion;
+    private ResultStorageEngine resultStorageEngine;
+    private String baseURLEndPoint;
+    private String downloadBasePath;
 
     @Override
-    public AsyncQueryResult call() throws NoHttpResponseException, URISyntaxException {
-         return processQuery();
+    public AsyncQueryResult call() throws URISyntaxException, IOException {
+        return processQuery();
     }
 
     public AsyncQueryThread(AsyncQuery queryObj, User user, Elide elide, QueryRunner runner,
-            AsyncQueryDAO asyncQueryDao, String apiVersion) {
+            AsyncQueryDAO asyncQueryDao, String apiVersion, ResultStorageEngine resultStorageEngine,
+            String baseURLEndPoint, String downloadBasePath) {
```

Comment:
These need different names (and maybe helpful comments to distinguish them).  I suggested requestURL and downloadPath or something along those lines.